02-library/cs506/sim.py

```
@@ -1,13 +1,41 @@
+import math
 def euclidean_dist(x, y):
-    raise NotImplementedError()
+    sum =0
+    for i, j in zip(x,y):
+        sum += (i-j)**2
+    return math.sqrt(sum)
 
 def manhattan_dist(x, y):
-    raise NotImplementedError()
+    sum = 0
+    for i ,j in zip(x, y):
+        sum += abs(i-j)
+    return sum
+
 
 def jaccard_dist(x, y):
-    raise NotImplementedError()
+    m = set(x)
+    n = set(y)
+    
+    try :
+        sim = len(n&m)/ (len(n)+len(m)- len(n&m))
+        return 1-sim
+    except ZeroDivisionError as e:
+        print(e)
```

you should raise an error earlier if x and y are empty